plugins/rich-editor/RichEditorPlugin.php

```diff
@@ -18,11 +19,17 @@ class RichEditorPlugin extends Gdn_Plugin {
     /** @var integer */
     private static $editorID = 0;
 
+    /** @var \Vanilla\Formatting\FormatService */
+    private $formatService;
+
     /**
      * Set some properties we always need.
+     *
+     * @param \Vanilla\Formatting\FormatService $formatService
      */
-    public function __construct() {
+    public function __construct(\Vanilla\Formatting\FormatService $formatService) {
         parent::__construct();
+        $this->formatService = $formatService;
```

Comment:
I am not sure about the order. I have a feeling that it is better to have it reordered.
```suggestion
        $this->formatService = $formatService;
        parent::__construct();
```

But not sure about `self::$editorID++;` part